packages/core/src/shader/ShaderPass.ts

```diff
@@ -94,6 +125,61 @@ export class ShaderPass extends ShaderPart {
       return shaderProgram;
     }
 
+    if (this._type === ShaderType.Canonical) {
+      return this._getCanonicalShaderProgram(engine, macroCollection);
+    }
```

Comment:
This is parallel logic. It is recommended to use else to enhance readability.

Reply:
done